source/main/gfx/GfxActor.cpp

```diff
@@ -919,16 +917,26 @@ void RoR::GfxActor::UpdateRods()
 
     for (Rod& rod: m_rods)
     {
-        Ogre::Vector3 pos1 = m_actor->ar_nodes[rod.rod_node1].AbsPosition;
-        Ogre::Vector3 pos2 = m_actor->ar_nodes[rod.rod_node2].AbsPosition;
+        int i = rod.rod_beam_index;
+        if (m_actor->ar_beams[i].bm_disabled || m_actor->ar_beams[i].bm_broken)
```

Comment:
You can't read beam state here, it's being updated from physics threadpool tasks while this code is running. And even if it wasn't true at the moment, it breaks the concept of separating simulation (fixed timestep) and async. scene update + rendering (framestep). We need to be strict about this, else we'll end up with very tricky MT bugs in the future.

Reply:
> You can't read beam state here

I know, I really should have added a TODO comment.